runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBox.java

```diff
@@ -206,6 +225,42 @@ private void rebuildAsync(boolean updated, XpSnapshotSingle xpSnapshotSingle)
 		expHour.setText(htmlLabel("XP/Hour: ", xpSnapshotSingle.getXpPerHour()));
 	}
 
+	int getKillsRemaining(XpSnapshotSingle xpSnapshotSingle)
+	{
+		Actor opponent = client.getLocalPlayer().getInteracting();
+		
+		if (opponent != null)
+		{
+			int opponentHealth = oppInfoHealth.get(opponent.getName() + "_" + opponent.getCombatLevel());
+			double modifier = getCombatXPModifier(skill);
+			double actionExp = (opponentHealth * modifier);
+			killsRemaining = (int) Math.ceil(xpSnapshotSingle.getXpRemainingToGoal() / actionExp);
+		}
+
+		return killsRemaining;
```

Comment:
So, if it is a combat skill, and the player used a lamp after logging in, they will see MAX_INT kills remaining?
And if they fought, but then used a lamp, their killsRemaining will be the same.

Try to fall back to actions remaining (and say actions instead of kills) when no opponent can be determined.